src/calendar/view/CalendarEventEditDialog.ts

```diff
@@ -581,6 +506,110 @@ export async function showCalendarEventDialog(
 	dialog.show()
 }
 
+function renderRepeatRulePicker(viewModel: CalendarEventViewModel, startOfTheWeekOffset: number): Children {
+	const intervalValues = createIntervalValues()
+	const endTypeValues = createRepeatRuleEndTypeValues()
+	return [
+		renderTwoColumnsIfFits(
+			[
+				// Repeat type == Frequency: Never, daily, annually etc
+				m(".flex-grow.pr-s", renderRepeatPeriod(viewModel)), // Repeat interval: every day, every second day etc
+				m(".flex-grow.pl-s" + (viewModel.repeat ? "" : ".hidden"), renderRepeatInterval(viewModel, intervalValues)),
+			],
+			viewModel.repeat
+				? [
+						m(".flex-grow.pr-s", renderEndType(viewModel, endTypeValues)),
+						m(".flex-grow.pl-s", renderEndValue(viewModel, intervalValues, startOfTheWeekOffset)),
+				  ]
+				: null,
+		),
+		viewModel.repeat && viewModel.repeat.excludedDates.length > 0
+			? // we're rendering a hidden element to make the delete exclusion text field the same size as the other elements in this section
+			  renderTwoColumnsIfFits([m(".flex-grow.pr-s", renderExclusionCount(viewModel)), m(".flex-grow.pl-s.hidden", lang.get("emptyString_msg"))], [])
+			: null,
+	]
+}
+
+function renderExclusionCount(viewModel: CalendarEventViewModel): Children {
+	return m(TextField, {
+		label: () => "Excluded Dates", //FIXME phrase
+		value: viewModel.repeat?.excludedDates.length.toString() ?? "0",
+		injectionsRight: () => renderDeleteExclusionButton(viewModel),
+		disabled: true,
+	})
+}
+
+function renderDeleteExclusionButton(viewModel: CalendarEventViewModel): Children {
+	return m(IconButton, {
+		title: () => "delete exclusions", //FIXME add to phrase
+		click: () => viewModel.deleteExcludedDates(),
+		icon: Icons.Trash,
+	})
+}
+
+function renderRepeatPeriod(viewModel: CalendarEventViewModel) {
+	const repeatValues: SelectorItemList<RepeatPeriod | null> = createRepeatRuleFrequencyValues()
+	return m(DropDownSelector, {
+		label: "calendarRepeating_label",
+		items: repeatValues,
+		selectedValue: (viewModel.repeat && viewModel.repeat.frequency) || null,
+		selectionChangedHandler: (period) => viewModel.onRepeatPeriodSelected(period),
+		icon: BootIcons.Expand,
+		disabled: viewModel.isReadOnlyEvent(),
+	} as DropDownSelectorAttrs<RepeatPeriod | null>)
```

Comment:
indeed

Reply:
turns out that at least tests don't compile with this because one of our transformations can't deal with this yet. postponing it until we get to updating them (`as` was used here before anyway)